modules/node/src/admin/admin.service.ts

```diff
@@ -202,4 +233,106 @@ export class AdminService {
 
     return toMerge;
   }
+
+  /**
+   * December 9, 2019
+   *
+   * We added a "proxyFactoryAddress" field to the state channels. This allows channels that were
+   * created at any time to be able to be deployed, even if the proxy factory address was changed.
+   *
+   * This function will fix old channels that did not have this field. The function does the
+   * following:
+   *
+   * 1. Calculate what the multisig address should be based on our "current" proxy factory address
+   *    for the specific network.
+   *   a. If it is correct, add the "proxyFactoryAddress" field in the DB.
+   *   b. If it is not correct, try calculcating the multisig address based on historical proxy
+   *      factory addresses we have deployed before. Once we find the correct address, update it
+   *      in the DB.
+   *   c. If none of the addresses are correct, add the channel to a "still broken" list and
+   *      reconcile offline.
+   */
+  async fixProxyFactoryAddresses(): Promise<FixProxyFactoryAddressesResponse> {
+    const fixProxyFactoryInCfCoreRecord = async (
+      repoPath: string,
+      pfAddress: string,
+    ): Promise<void> => {
+      logger.log(`Multisig address is as expected, adding correct proxyFactory address`);
+      const cfCoreRecord = await this.cfCoreRepository.get(repoPath);
+      cfCoreRecord["proxyFactoryAddress"] = pfAddress;
+      await this.cfCoreRepository.set([
+        {
+          path: repoPath,
+          value: cfCoreRecord,
+        },
+      ]);
+    };
+
+    const fixedChannels = [];
+    const stillBrokenChannels = [];
+    const channels = await this.channelService.findAll();
+
+    for (const channel of channels) {
+      const { data: stateChannel } = await this.cfCoreService.getStateChannel(
+        channel.multisigAddress,
+      );
+      const contractAddresses = await this.configService.getContractAddresses();
+
+      if (stateChannel.multisigAddress !== channel.multisigAddress) {
+        // things will be very weird if this happens, so just dont allow this to happen
+        logger.error(
+          `This should never happen! stateChannel.multisigAddress ${stateChannel.multisigAddress} !== channel.multisigAddress ${channel.multisigAddress}`,
+        );
+        continue;
+      }
```

Comment:
wait -- this does happen in some cases where the multisig address changed and nothing stopped it from getting written to the store.

this is actually the exact check i used manually to verify *if* the multisig address changed over the lifetime of the channel

Reply:
Well we should have fixed all of them by now, so if we didn't let's blow up and fix it first!